json-rpc/tests/node.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use config_builder::BuildSwarm;
+use libra_config::config::NodeConfig;
+use libra_genesis_tool::config_builder::ValidatorBuilder;
+use std::process::{Child, Command};
+
+pub struct Node {
+    node: Child,
+    pub config: NodeConfig,
+    pub root_key: libra_crypto::ed25519::Ed25519PrivateKey,
+}
+
+impl Drop for Node {
+    // When the LibraNode struct goes out of scope we need to kill the child process
+    fn drop(&mut self) {
+        // check if the process has already been terminated
+        match self.node.try_wait() {
+            // The child process has already terminated, perhaps due to a crash
+            Ok(Some(_)) => {}
+
+            // The node is still running so we need to attempt to kill it
+            _ => {
+                if let Err(e) = self.node.kill() {
+                    panic!("LibraNode process could not be killed: '{}'", e);
+                }
+            }
+        }
+    }
+}
+
+impl Node {
+    pub fn start(node_dir: &std::path::Path) -> Result<Self> {
+        let config_path = node_dir.join("node.yaml");
+        let struct_log_path = node_dir.join("struct.log");
+        let log_path = node_dir.join("log");
+
+        // TODO: test with default_for_validator_full_node?
+        let builder = ValidatorBuilder::new(1, NodeConfig::default_for_validator(), node_dir);
```

Comment:
@davidiw your favorite config builder

Reply:
is there other way to config node? I'm fine with whatever works, this is the only way I found to configure a node.

Reply:
I misread this, I think the genesis_tool::config_builder is the right way to go. I thought it uses the deprecated one because of the added dependency
```
config-builder = { path = "../config/config-builder", version = "0.1.0" }
```

Reply:
I also refactored this code in an uncommitted PR, in the future @xli can just use libra_genesis_tool::test_config() and be done with it: https://github.com/libra/libra/pull/6072